integration/test/Testlib/MockIntegrationService.hs

```diff
@@ -95,10 +95,12 @@ withFreePortAnyAddr = bracket openFreePortAnyAddr (liftIO . Socket.close . snd)
 openFreePortAnyAddr :: MonadIO m => m (Warp.Port, Socket)
 openFreePortAnyAddr = liftIO $ bindRandomPortTCP (fromString "*")
 
+type LiftedApplication = Request -> (Wai.Response -> App ResponseReceived) -> App ResponseReceived
```

Comment:
it's nice to have access to the context of the surrounding application like the `Reader` and the state in the `IORef`s